build.sbt

```diff
@@ -11,12 +11,13 @@ lazy val root = (project in file(".")).
     )),
     name := "scalatest-example",
     Compile / PB.targets := Seq(
-      scalapb.gen() -> (Compile / sourceManaged).value / "scalapb"
+      scalapb.gen(FlatPackage) -> (Compile / sourceManaged).value / "scalapb",
+      PB.gens.java -> (Compile / sourceManaged).value / "java"
     ),
     Compile / PB.targets ++= Seq(
       Target(
         scalapb.validate.gen(FlatPackage),
-        (Compile / sourceManaged).value
+        (Compile / sourceManaged).value / "scalapb"
```

Comment:
`scalapb.gen` and `scalapb.validate.gen` also have to generate into the same directory, so the latter can update files generated by the former.